server/game/core/Player.js

```diff
@@ -174,6 +168,14 @@ class Player extends GameObject {
         return this.getArenaCards(arena).filter((card) => card.isUnit() && card !== ignoreUnit && card.hasSomeAspect(aspect) && cardCondition(card));
     }
 
+    /**
+     * @param { String } name the name of the unit to check for control of
+     * @returns { boolean } true if this player controls a unit with the given name
+     */
+    controlsUnitWithName(name) {
```

Comment:
This name is a little misleading imo since it counts the leader even if it's not currently a unit (i.e. not currently deployed).